### PR TITLE
Raise an error when trying to use SD Cascade Decoder with dtype bfloat16 and torch < 2.2

### PR DESCRIPTION
**Pull request by DN6**
# What does this PR do?
It is currently not possible to use the `bfloat16` dtype with the SD Cascade decoder pipeline when torch version < 2.2
This PR adds a check that raises an error when this is attempted. 

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/diffusers/pipelines/stable_cascade/pipeline_stable_cascade.py [hunk 1]
```diff
@@ -361,6 +361,8 @@ def __call__(
         device = self._execution_device
         dtype = self.decoder.dtype
         self._guidance_scale = guidance_scale
+        if is_torch_version("<", "2.2.0") and dtype == torch.bfloat16:
```
Comment (sayakpaul) [inline src/diffusers/pipelines/stable_cascade/pipeline_stable_cascade.py:364]: Is it only the decoder dtype or the general `torch_dtype` that we use?

**Review by sayakpaul — APPROVED** [review]
LGTM. 

Would be great if we added a test here as well to make sure this bug is raised accordingly.

**Comment by DN6** [comment]
> Would be great if we added a test here as well to make sure this bug is raised accordingly.

Maybe in a follow up? We would need a new env with torch 2.1, all our containers use 2.2 now.

**Comment by sayakpaul** [comment]
Yeah no need then. Feel free to merge!

**Pull request merged**